tests/unit/test_world_scientific.py

```diff
@@ -13,52 +13,44 @@
 import os
 
 from scrapy.crawler import Crawler
+from scrapy.http import TextResponse
 
 from hepcrawl.pipelines import InspireCeleryPushPipeline
 from hepcrawl.spiders import wsp_spider
 
 from .responses import fake_response_from_file
 
 
-@pytest.fixture
-def results():
-    """Return results generator from the WSP spider."""
-    spider = wsp_spider.WorldScientificSpider()
-    records = list(spider.parse(
-        fake_response_from_file('world_scientific/sample_ws_record.xml')
-    ))
-    assert records
-    return records
-
-
-@pytest.fixture
-def spider():
+def create_spider():
     crawler = Crawler(spidercls=wsp_spider.WorldScientificSpider)
     return wsp_spider.WorldScientificSpider.from_crawler(crawler)
 
 
-@pytest.fixture
-def one_result(spider):
-    """Return results generator from the WSP spider. Tricky fields, one
-    record.
-    """
-    from scrapy.http import TextResponse
-    
+def get_results(file_name):
+    """Return all results generator from the WSP spider via pipelines."""
     # environmental variables needed for the pipelines payload
     os.environ['SCRAPY_JOB'] = 'scrapy_job'
     os.environ['SCRAPY_FEED_URI'] = 'scrapy_feed_uri'
     os.environ['SCRAPY_LOG_FILE'] = 'scrapy_log_file'
 
-    record = spider.parse(
+    spider = create_spider()
+    records = spider.parse(
         fake_response_from_file(
-            'world_scientific/wsp_record.xml',
+            file_name=file_name,
             response_type=TextResponse
         )
     )
 
     pipeline = InspireCeleryPushPipeline()
     pipeline.open_spider(spider)
-    return pipeline.process_item(record.next(), spider)
+
+    return (pipeline.process_item(record, spider) for record in records)
+
+
+def get_one_result(file_name):
+    """Return one record generator from the WSP spider via pipelines."""
```

Comment:
bad docs, actually, it's so simple and well named that we don't need the docs :)